src/http/api/v1/matches.ts

```diff
@@ -0,0 +1,120 @@
+import express, { Response, Request } from "express";
+
+const upload = multer();
+import multer from "multer";
+import ClientError from "../clientError.js";
+
+import Calculator, {
+  Overrides,
+  DataProvider,
+  FileSystemDataProvider,
+  CalculatorOptions,
+  FileNotFoundError,
+  ResourceNotFoundError,
+  parseOverrides,
+  CalculatorError,
+} from "../../../calculator/index.js";
+
+const router = express.Router();
+
+function handleError(err: unknown) {
+  if (err instanceof FileNotFoundError) {
+    throw new ClientError(err.message, 404);
+  }
+
+  if (err instanceof ResourceNotFoundError) {
+    throw new ClientError(err.message, 404);
+  }
+
+  if (err instanceof CalculatorError) {
+    throw new ClientError(err.message, 400);
+  }
+
+  // unexpected error, rethrow to upper level handler
+  throw err;
+}
+
+export const calculatorConfig: { dataProvider: DataProvider } = {
+  dataProvider: new FileSystemDataProvider("./data"),
+};
+
+async function matchesHandler(
+  req: Request,
```

Comment:
again, totally unnecessary, but possible - we can narrow down the generic `Request` type to be _really_ specific (this goes for _all_ the express Request/Response types. 
```typescript
type MatchRequest = Request<
  { chainId: string; roundId: string },
  any,
  any,
  {
    minimumAmount?: string;
    passportThreshold?: string;
    matchingCapAmount?: string;
    enablePassport?: string;
    ignoreSaturation?: string;
  }
>;
```
such that 
```
async function matchesHandler(
  req: MatchRequest,
  ...
```
etc, etc. again just nitpicking :)